blinky-statik/src/main/java/org/spideruci/analysis/db/DatabaseWriter.java

```diff
@@ -0,0 +1,98 @@
+package org.spideruci.analysis.db;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.spideruci.analysis.statik.flow.data.SourceICFGNode;
+
+public class DatabaseWriter {
+	private static Connection c = null;
+	private static Map<SourceICFGNode, Integer> nodeIdMap = new HashMap<>();
+	private static int nodeId = 0; 
+
+	public DatabaseWriter(Connection c) {
+		this.c = c;
+	}
+	
+	private int getNodeId(SourceICFGNode node){
+
+		int id;
+		if(nodeIdMap.get(node) == null){
+			id = nodeId++;
+			nodeIdMap.put(node, id);
+		}
+		else id = nodeIdMap.get(node);
```

Comment:
ideally, wrap `id = nodeIdMap.get(node);` in curly braces as well, make it consistent with the usage of curly braces in the if-statement above.